fbc-test.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env sh
+
+trap teardown EXIT
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+DEV_KUBECONFIG="--kubeconfig=$SCRIPTDIR/stage_dev_release_kubelogin"
+MANAGED_KUBECONFIG="--kubeconfig=$SCRIPTDIR/stage_managed_release_kubelogin"
+
+MANAGED_NAMESPACE="managed-release-team-tenant"
+APPLICATION_NAME="e2e-fbc-application"
+COMPONENT_NAME="e2e-fbc-component"
+RELEASE_PLAN_NAME="e2e-fbc-releaseplan"
+RELEASE_PLAN_ADMISSION_NAME="e2e-fbc-releaseplanadmission"
+RELEASE_STRATEGY_NAME="e2e-fbc-strategy"
+TIMEOUT_SECONDS=600
+
+function setup() {
+    
+    
+    echo "Creating Application"
+    kubectl apply -f release-resources/application.yaml "$DEV_KUBECONFIG"
+
+    echo "Creating Component"
+    kubectl apply -f release-resources/component.yaml "$DEV_KUBECONFIG"
+    
+    echo "Creating ReleaseStrategy"
+    kubectl apply -f release-resources/release-strategy.yaml "$MANAGED_KUBECONFIG"
+
+    echo "Creating ReleasePlan"
+    kubectl apply -f release-resources/release-plan.yaml "$DEV_KUBECONFIG"
+
+    echo "Creating ReleasePlanAdmission"
+    kubectl apply -f release-resources/release-plan-admission.yaml "$MANAGED_KUBECONFIG"
+
+}
+
+function teardown() {
+   
+    kubectl delete pr -l "appstudio.openshift.io/application="$APPLICATION_NAME",pipelines.appstudio.openshift.io/type="$type",appstudio.openshift.io/component="$COMPONENT_NAME"" "$DEV_KUBECONFIG"
+    kubectl delete pr -l "appstudio.openshift.io/application="$APPLICATION_NAME",pipelines.appstudio.openshift.io/type="$type"" "$MANAGED_KUBECONFIG"
+    kubectl delete release "$DEV_KUBECONFIG" -o=jsonpath='{.items[?(@.spec.releasePlan=="$RELEASE_PLAN_NAME")].metadata.name}'
+    kubectl delete releaseplan "$RELEASE_PLAN_NAME" "$DEV_KUBECONFIG"
+    kubectl delete releaseplanadmission "$RELEASE_PLAN_ADMISSION_NAME" "$MANAGED_KUBECONFIG"
+    kubectl delete releasestrategy "$RELEASE_STRATEGY" "$MANAGED_KUBECONFIG"
+
+    if kubectl get application "$APPLICATION_NAME"  "$DEV_KUBECONFIG" &> /dev/null; then
+        echo "Application '"$APPLICATION_NAME"' exists. Deleting..."
```

Comment:
If this happens we have a bug in the release-service

Reply:
How is so? We don't delete applications. We don't even delete release when an application is deleted.

Reply:
Look above. We delete the releaseplan. The releaseplan owns the application doesn't it? Or do I have it backwards? Either way, one should cause the other to be deleted. We shouldn't have to delete them both

Reply:
agree with David, 
we van delete the Release without an application can be deleted 
so as part of the tear down we should delete the application an component.

Reply:
if we delete the application, the releaseplan is deleted. so why manually delete the releaseplan?

Reply:
This is right after latest change was made, so I agree and I'll remove